helix-term/src/ui/editor.rs

```diff
@@ -224,10 +224,7 @@ impl EditorView {
 
         Self::render_diagnostics(doc, view, inner, surface, theme);
 
-        let statusline_area = view
-            .area
-            .clip_top(view.area.height.saturating_sub(1))
-            .clip_bottom(1); // -1 from bottom to remove commandline
+        let statusline_area = view.area.clip_top(view.area.height);
```

Comment:
This introduces the issue with split views. The easiest solution I could come up with was something like this:

```rust
// if we're not at the bottom of the screen, draw the statusline higher, otherwise
// it'll overlap with the next view
let statusline_area = if (viewport.bottom() - 1) == view.area.bottom() {
    view.area.clip_top(view.area.height)
} else {
    view.area
        .clip_top(view.area.height.saturating_sub(1))
        .clip_bottom(1)
};
```